Allow quests to be completed out of order

AlyxMoon opened the pull request:
Fixes #9461

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
Updated file `website\server\models\group.js`
- Changed the check that current completed quest is the final in Master Classer series, it now checks that the completed quest is any in the series
- Updated query to include last Master Classer quest

Updated test file `test\api\v3\unit\models\group.test.js`
- Added a new test to check whether Master Classer achievement is given when the final quest to be completed is not the final one in the series

----
UUID: cc212fcc-5f96-403b-a785-17ba66c7b11a

AlyxMoon committed:
Allow quests to be completed out of order

Changes check in website/server/models/group.js to check for any of the
Master Classer Quests, not just the last one. Updated query to check
that all quests are done before giving achievement.

Fixes #9461

AlyxMoon committed:
Allow quests to be completed out of order

Changes check in website/server/models/group.js to check for any of the
Master Classer Quests, not just the last one. Updated query to check
that all quests are done before giving achievement.

Fixes #9461

TheHollidayInn committed:
Added account transfer migration (#9548)

* Added account transfer migration

* Removed bad comment

AlyxMoon commented:
I have no idea how I reduced test coverage that much, but clearly I have some edits to make 😕

TheHollidayInn committed:
Should do diff week fix (#9551)

* Fixed week difference when changing year

* Added year switchover test

* Fixed start date setting

TheHollidayInn committed:
Cancel users free group plan when they leave a group (#9543)

* Cancel users free group plan when they leave a group

* Fixed lint

TheHollidayInn committed:
Fixed display of RYA behind bailey (#9555)

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
4.12.1

SabreCat committed:
Merge branch 'release' into develop

paglias commented on test/api/v3/unit/models/group.test.js:
```diff
@@ -1418,6 +1418,42 @@ describe('Group Model', () => {
+          updatedLeader,
+          updatedParticipatingMember,
+        ] = await Promise.all([
+          User.findById(questLeader._id),
```
findById() should be followed by `.exec()` otherwise it returns an object that has a `.then` but is not a realy promise, see http://mongoosejs.com/docs/api.html#model_Model.findById

AlyxMoon replied:
Okay, thanks for the feedback. Got that changed.

paglias reviewed (changes requested):
One small change but otherwise it looks good :) Don't worry about the coverage, it's a bit buggy

paglias committed:
fix navbar not showing up in homepage

Alys committed:
remove words related to alcohol

This is because they're causing problems in housework guilds (alcohol
is used for cleaning) and since it's coming up to Christmas a lot of
guilds are starting to have permissible conversations about drinks,
rum balls, etc.

We might put these back after Christmas but make specific exceptions
for the housework guilds.

When a more advanced word blocker is introduced, we'll be able to
ban alcohol words only in the Tavern. For irony.

AlyxMoon committed:
Add call to exec() after findById() in tests

AlyxMoon committed:
Add call to exec() after findById() in tests

paglias commented:
@AlyxMoon it looks like there is a missing comma that is preventing the lint test from passing https://travis-ci.org/HabitRPG/habitica/jobs/306918838#L907, and that some tests broke after this change https://travis-ci.org/HabitRPG/habitica/jobs/306918839

AlyxMoon committed:
Fix the dangling comma

AlyxMoon committed:
Fix the dangling comma

AlyxMoon committed:
Add `exec()` call to first masterClasser test

Since the TravisCI test was failing on my new test and was fixed by
adding an `exec()` call, did the same to existing test 'gives out super
awesome Masterclasser achievement to the deserving'

AlyxMoon committed:
Merge branch 'Allow-quests-to-be-completed-in-any-order-for-achievement' of https://github.com/AlyxMoon/habitica into Allow-quests-to-be-completed-in-any-order-for-achievement

AlyxMoon committed:
Remove mistakenly added extra test

AlyxMoon commented:
So I feel a bit silly about writing so many commits that didn't do much.

I had to fix some stuff on my end just to get the linting/tests to run without errors, and now everything is working out okay on my end with the tests passing. Are the tests on travis-ci different? I honestly can't figure out what to fix with my code at this point.

paglias commented:
@AlyxMoon I think something went wrong with git because if you look at the changes in this PR it now includes a lot of extraneous changes. If you want to use a new clean branch you can pick specific commits from this PR with `git cherry-pick COMMIT_HASH` while in the new branch
